src/filter/range.rs

```diff
@@ -7,18 +8,22 @@ where
     for<'a> K: Key<'a>,
 {
     #[serde(serialize_with = "serialize_key", deserialize_with = "deserialize_key")]
-    min: K,
+    min: Arc<RwLock<K>>,
     #[serde(serialize_with = "serialize_key", deserialize_with = "deserialize_key")]
-    max: K,
-    initialized: bool,
+    max: Arc<RwLock<K>>,
+    #[serde(
+        serialize_with = "serialize_initialized",
+        deserialize_with = "deserialize_initialized"
+    )]
+    initialized: Arc<RwLock<bool>>,
```

Comment:
Three `RwLock<>` is too much. You should split `RangeFilter<>` into 2 structs:
1. `RangeFilterSafe<>` that assumes to work under protection (basically it is the copy of the original `RangeFilter<>`)
2. `RangeFilter<>` that contains only one field: `RwLock<RangeFilterSafe<>>` and redirects all operations to `RangeFilterSafe<>`